### PR TITLE
[Bugfix] Fix potential race condition in ColumnReader

### LINKED ISSUE #6762: Potential race condition in ColumnReader
The following methods of `ColumnReader` may modify the `ColumnReader::_flags` in parallel:
```
    Status _load_zone_map_index(bool use_page_cache, bool kept_in_memory);
    Status _load_ordinal_index(bool use_page_cache, bool kept_in_memory);
    Status _load_bitmap_index(bool use_page_cache, bool kept_in_memory);
    Status _load_bloom_filter_index(bool use_page_cache, bool kept_in_memory);
```

### PR DESCRIPTION
**Pull request by sduzh**
## What type of PR is this：
- [x] bug
- [ ] feature
- [ ] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #6762

### CODE REVIEW

File: be/src/storage/rowset/bitmap_index_reader.h [hunk 1]
```diff
@@ -69,13 +69,22 @@ class BitmapIndexReader {
...
 private:
     friend class BitmapIndexIterator;
 
+    Status do_load(FileSystem* fs, const std::string& filename, const BitmapIndexPB& meta, bool use_page_cache,
+                   bool kept_in_memory);
+
+    // 0: data has not been loaded
+    // 1: loading in process
+    // 2: data has been load in memory
+    std::atomic<int> _state;
```
Comment (imay) [inline be/src/storage/rowset/bitmap_index_reader.h:93]: how about using an enum

File: be/src/storage/rowset/bitmap_index_reader.cpp [hunk 2]
```diff
@@ -21,22 +21,51 @@
...
-    _typeinfo = get_type_info(OLAP_FIELD_TYPE_VARCHAR);
-    const IndexedColumnMetaPB& dict_meta = bitmap_index_meta->dict_column();
-    const IndexedColumnMetaPB& bitmap_meta = bitmap_index_meta->bitmap_column();
-    _has_null = bitmap_index_meta->has_null();
+StatusOr<bool> BitmapIndexReader::load(FileSystem* fs, const std::string& filename, const BitmapIndexPB& meta,
+                                       bool use_page_cache, bool kept_in_memory) {
+    while (true) {
+        auto state = _state.load();
+        if (state == 2) {
+            return false;
```
Comment (imay) [inline be/src/storage/rowset/bitmap_index_reader.cpp:38]: why return false? should be true/

Comment (sduzh) [reply be/src/storage/rowset/bitmap_index_reader.cpp:38]: 'false' means the index data was loaded by another caller.

File: be/src/storage/rowset/bitmap_index_reader.h (hunk 1 above)
Comment (sduzh) [reply be/src/storage/rowset/bitmap_index_reader.h]: Updated, PTAL

**Review by chaoyli — APPROVED** [review]

**Review by imay — APPROVED** [review]

**Pull request merged**